### PR TITLE
Fix: #925. Example with PUT+204.

### LINKED ISSUE #925: Add example with PUT + 204
## I expect

an example digest of PUT  + 204

## Note 

PUT without complete representation
If PUT returns 204 it may return a Digest header, provided that the response contains all the required representation metadata.

```
PUT /books/123 HTTP/1.1
Content-Type: application/json

{"id": "123", "title": "Foo"}

---

HTTP/1.1 204 No Content
Content-Type: application/json
Digest: sha256=TtjhvKObwiGt3XI75TExBCKFSkySLPWqQptGn190rMQ=
```

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

Adds an example with PUT + 204 No Content.

Response contains a Digest computed on the selected representation.

Related to #853
Fixes #925

### CODE REVIEW

**Comment by ioggstream** [comment]
Hi @reschke, your feedback on this would be great as this is related to https://github.com/httpwg/http-core/issues/255

**Comment by ioggstream** [comment]
> Questions:
> 
> 1. is the Digest in the request supposed to match the payload? (it doesn't seem to, but maybe I'm doing something wrong)

I used the following (the request uses the `identity` content coding, the response uses `br`
```
python3 -c '
import base64, json, hashlib
s = json.dumps(dict(hello="world")).encode()
print(s)

h = hashlib.sha256(s).digest()
print(base64.encodebytes(h))
'
```

If it's not ok let me know and I'll fix it.

> 2. please put request and response into different artwork sections

```
Request:

~~~
...

~~~


Response:

~~~
...

~~~

```

thanks++

**Comment by reschke** [comment]
Maybe it's the JSON serializer. I tried

~~~
echo -n '{"hello": "world"}' | od -cx
0000000   {   "   h   e   l   l   o   "   :       "   w   o   r   l   d
           227b    6568    6c6c    226f    203a    7722    726f    646c
0000020   "   }
           7d22
0000022
~~~

~~~
echo -n '{"hello": "world"}' | sha256sum.exe | cut -d ' ' -f1
5f8f04f6a3a892aaabbddb6cf273894493773960d4a325b105fee46eef4304f1
~~~

~~~
echo -n '{"hello": "world"}' | sha256sum.exe | cut -d ' ' -f1 | tr -d "\n" | base64
NWY4ZjA0ZjZhM2E4OTJhYWFiYmRkYjZjZjI3Mzg5NDQ5Mzc3Mzk2MGQ0YTMyNWIxMDVmZWU0NmVl
ZjQzMDRmMQ==
~~~

**Comment by ioggstream** [comment]
@reschke it seems you're base64-encoding the `hexdigest` instead of the `bytes`.

> echo -n '{"hello": "world"}' | sha256sum.exe | cut -d ' ' -f1
> 5f8f04f6a3a892aaabbddb6cf273894493773960d4a325b105fee46eef4304f1

Your output is the one of:

```
python>>>hex_digest = h.hexdigest().encode()
b'5f8f04f6a3a892aaabbddb6cf273894493773960d4a325b105fee46eef4304f1'

python>>>base64.encodebytes(hex_digest)                                                                                                                                                                                         
b'NWY4ZjA0ZjZhM2E4OTJhYWFiYmRkYjZjZjI3Mzg5NDQ5Mzc3Mzk2MGQ0YTMyNWIxMDVmZWU0NmVl\nZjQzMDRmMQ=='
```

While afaik you should 
```
>>> byte_digest = h.digest()                                                                                                                                                                                                                         
b'_\x8f\x04\xf6\xa3\xa8\x92\xaa\xab\xbd\xdbl\xf2s\x89D\x93w9`\xd4\xa3%\xb1\x05\xfe\xe4n\xefC\x04\xf1'
>>> base64.encode(byte_digest)
X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=
```

Does it sound right to you?

**Comment by reschke** [comment]
Indeed:

~~~
$ echo -n '{"hello": "world"}' | sha256sum| cut -d ' ' -f1|xxd -r -p|base64
X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=
~~~

**Comment by ioggstream** [comment]
> 1. I think you should actually state that the Digest in the request is for the enclosed payload.

Done

> 2. The artwork should match the actual HTTP/1.1 request; so no leading and trailing blank lines (the latter would imply that there is a line break in the payload).

I'll address all other occurencies in a new PR. Thanks!

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -592,6 +592,35 @@ Response:
 
 ~~~
 
+
+### Digest in both Request and Response. Representation data is not contained in the response payload
+
+Request `Digest` value is calculated on the enclosed payload.
+Returned `Digest` value depends on the representation metadata header fields,
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md]:
```suggestion
Response `Digest` value depends on the representation metadata header fields,
```

just for consistency

**Comment by ioggstream** [comment]
:heart: @reschke @LPardue for all the guidance!

**Pull request merged**